src/FakeItEasy/Configuration/AfterMustHaveHappenedConfiguration.cs

```diff
@@ -17,11 +17,38 @@ public AfterMustHaveHappenedConfiguration(FakeManager fakeManager, ICallMatcher
             this.repeatConstraint = repeatConstraint;
         }
 
-        public void InOrder(ISequentialCallContext context)
+        /// <summary>
+        /// Checks that the asserted call happened in order relative to others in the assertion chain.
+        /// </summary>
+        /// <param name="nextAssertion">An assertion describing the next call that should occur.</param>
+        /// <returns>An object that can be used to assert that a following call was made in the expected order.</returns>
+        /// <exception cref="ExpectationException">The call was not made in the expected order.</exception>
+        public IOrderableCallAssertion Then(IAfterMustHaveHappenedConfiguration nextAssertion)
         {
-            Guard.AgainstNull(context, "context");
+            var context = ServiceLocator.Current.Resolve<SequentialCallContext>();
+            this.CheckCallHappenedInOrder(context);
+            return new SubsequentOrderedCallAssertion(context).Then(nextAssertion);
+        }
 
+        private void CheckCallHappenedInOrder(SequentialCallContext context)
+        {
             context.CheckNextCall(this.fakeManager, this.matcher.Matches, this.callDescription, this.repeatConstraint);
         }
+
+        private class SubsequentOrderedCallAssertion : IOrderableCallAssertion
+        {
+            private readonly SequentialCallContext context;
+
+            public SubsequentOrderedCallAssertion(SequentialCallContext context)
+            {
+                this.context = context;
+            }
+
+            public IOrderableCallAssertion Then(IAfterMustHaveHappenedConfiguration nextCall)
+            {
+                ((AfterMustHaveHappenedConfiguration)nextCall).CheckCallHappenedInOrder(this.context);

```

Comment:
@thomasleveque, I'd also not considered people extending the interface. I'm glad you brought up Liskov. It's an excellent point.
So, unless we want to broaden the interface, it does seem that returning a class it the way to go. It's not common throughout FakeItEasy's API, but we have an uncommon situation here. Although I could try to play around with the code a little more. Maybe I've just not found the best implementation. In the meantime:

Your idea of returning an abstract class with protected constructor (in order to allow clients to extend the class, I assume) should resolve the problem.
At least if someone _did_ extend the class and pass an instance to `Then`, then we'd end up calling the internal, non-virtual `CheckCallHappenedInOrder`, which is fine, but in this case, I wonder why we'd provide an extensible type anyhow. There seems to be no benefit for users to provide an extension. Or do you see one?

I'm happy to the name. I just kept `AfterMustHaveHappenedConfiguration` from the old implementation because it didn't bother me, and the class did about the same thing - it seemed that providing a `Then` method is not so different from providing an `InOrder`. 

Your suggested names are serviceable, but I'd consider a different tack. One reason for the class to be exposed is to distinguish it from the result of the `Then` to avoid the nested `Then`s. While thinking of that aspect, I thought of `UnorderedCallAssertion`, which a raw `MustHaveHappened` is. It's only by wrapping one in a `Then` or using the `Then` method on one that we come up with an ordered assertion. Then the class that I've called `SubsequentOrderedCallAssertion` could drop the `Subsequent`: once we start returning from `Then`, we've imposed an ordering.